### PR TITLE
Terraform 0.13 support

### PR DESCRIPTION
**Pull request by yupwei68**
Extend https://github.com/Azure/terraform-azurerm-aks/pull/45:
add variable `vnet_subnet_id`

Extend https://github.com/Azure/terraform-azurerm-aks/pull/25:
add variable `os_disk_size_gb`

Extend https://github.com/Azure/terraform-azurerm-aks/pull/8:
add variable `enable_http_application_routing`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +56,8 @@ module "aks" {
   client_id           = "your-service-principal-client-appid"
   client_secret       = "your-service-principal-client-password"
   prefix              = "prefix"
+
+  depends_on = [azurerm_resource_group.example]
```
Comment (njuCZ) [inline README.md:60]: terraform 0.12 does not support this field. is it added by mistake?

File: README.md [hunk 2]
```diff
@@ -3,7 +3,42 @@
...
+}
+
+module "network" {
+  source              = "Azure/network/azurerm"
+  resource_group_name = azurerm_resource_group.example.name
+  address_space       = "10.0.0.0/16"
+  subnet_prefixes     = ["10.0.1.0/24"]
+  subnet_names        = ["subnet1"]
+  depends_on          = [azurerm_resource_group.example]
+
```
Comment (njuCZ) [inline README.md:25]: move the empty line to top of "depends_on"

File: README.md [hunk 3]
```diff
@@ -3,7 +3,42 @@
...
+module "aks" {
+  source              = "Azure/aks/azurerm"
+  resource_group_name = azurerm_resource_group.example.name
+  client_id           = "your-service-principal-client-appid"
+  client_secret       = "your-service-principal-client-password"
+  prefix              = "prefix"
+  vnet_subnet_id      = module.network.vnet_subnets[0]
+  os_disk_size_gb     = 50
+
+  depends_on = [azurerm_resource_group.example]
```
Comment (njuCZ) [inline README.md:37]: I think it's better to specify `depends_on = [module.network]`

File: README.md [hunk 4]
```diff
@@ -3,7 +3,41 @@
...
+module "aks" {
+  source              = "Azure/aks/azurerm"
+  resource_group_name = azurerm_resource_group.example.name
+  client_id           = "your-service-principal-client-appid"
+  client_secret       = "your-service-principal-client-password"
+  prefix              = "prefix"
+  vnet_subnet_id      = module.network.vnet_subnets[0]
+  os_disk_size_gb     = 50
+
+  depends_on = [azurerm_resource_group.example, module.network]
```
Comment (njuCZ) [inline README.md:36]: only "module.network" is needed, because "module.network" depends on "azurerm_resource_group.example"

File: main.tf [hunk 5]
```diff
@@ -26,18 +26,23 @@ resource "azurerm_kubernetes_cluster" "main" {
     name            = "nodepool"
     node_count      = var.agents_count
     vm_size         = var.agents_size
-    os_disk_size_gb = 50
+    os_disk_size_gb = var.os_disk_size_gb
+    vnet_subnet_id  = var.vnet_subnet_id
```
Comment (njuCZ) [inline main.tf:30]: if users don't specify "vnet_subnet_id" when using this module, it defaults to null, will there be some errors ?

File: variables.tf [hunk 6]
```diff
@@ -54,4 +64,22 @@ variable "enable_log_analytics_workspace" {
...
+variable "os_disk_size_gb" {
+  description = "Disk size of nodes in GBs."
+  type        = number
+  default     = 50
+}
+
+variable "enable_http_application_routing" {
+  description = "Enable HTTP Application Routing Addon (forces recreation)."
+  type        = bool
+  default     = false
```
Comment (njuCZ) [inline variables.tf:84]: it seems adding those extra fields is to serve another task, could we split into different PR?

Comment (yupwei68) [reply variables.tf:84]: I normally to merge functions to one PR to serve one release.

File: main.tf (hunk 5 above)
Comment (yupwei68) [reply main.tf:30]: Not yet. null is accepted for `vnet_subnet_id`, I have just tested it.

**Comment by yupwei68** [comment]
@njuCZ Thanks for comments. Changes have been pushed. Please continue reviewing.

**Review by njuCZ — APPROVED** [review]
